ibc_public/utils_connectivity.py

```diff
@@ -0,0 +1,141 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from nilearn import datasets, plotting, connectome
+import pandas as pd
+import os
+import seaborn as sns
+from scipy import stats
+from scipy.signal import correlate2d
+from ibc_public.utils_data import get_subject_session, DERIVATIVES
+from glob import glob
+
+
+def get_ses_modality(task):
+    """Get session numbers and modality for given task
+
+    Parameters
+    ----------
+    task : str
+        name of the task
+
+    Returns
+    -------
+    sub_ses : dict
+        dictionary with subject as key and session number as value
+    modality : str
+        modality of the task
+    """
+    if task == "GoodBadUgly":
+        # session names with movie task data
+        session_names = ["BBT1", "BBT2", "BBT3"]
+    elif task == "MonkeyKingdom":
+        # session names with movie task data
+        session_names = ["monkey_kingdom"]
+    elif task == "Raiders":
+        # session names with movie task data
+        session_names = ["raiders1", "raiders2"]
+    elif task == "RestingState":
+        # session names with RestingState state task data
+        session_names = ["mtt1", "mtt2"]
+    elif task == "DWI":
+        # session names with diffusion data
+        session_names = ["anat1"]
+    # get session numbers for each subject
+    subject_sessions = sorted(get_subject_session(session_names))
+    sub_ses = {}
+    if task == "DWI":
+        modality = "structural"
+        sub_ses = {
+            subject_session[0]: "ses-12"
+            if subject_session[0] in ["sub-01", "sub-15"]
+            else subject_session[1]
+            for subject_session in subject_sessions
+        }
+    else:
+        modality = "functional"
+        for subject_session in subject_sessions:
+            if (
+                subject_session[0] in ["sub-11", "sub-12"]
+                and subject_session[1] == "ses-13"
+            ):
+                continue
+            try:
+                sub_ses[subject_session[0]]
+            except KeyError:
+                sub_ses[subject_session[0]] = []
```

Comment:
Why do we need this ?

Reply:
I added some comments in the script to clarify this better